core/src/main/resources/lib/form/file.jelly

```diff
@@ -0,0 +1,51 @@
+<!--
+The MIT License
+
+Copyright (c) 2022, Jenkins project contributors
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:f="/lib/form">
+  <st:documentation>
+    Generates an input field <code>&lt;input type="file" ... /></code>
+
+    <st:attribute name="field">
+      Used for databinding.
+    </st:attribute>
+    <st:attribute name="jsonAware">
+      Enable structured form submission.
+    </st:attribute>
+    <st:attribute name="accept">
+      Defines the file types the file input should accept. This string is a comma-separated list.
```

Comment:
Optional I guess?

You could just copy unknown tag attrs to the HTML tag. There is some Jelly/Stapler trick to do that.

Reply:
Yes, `use="required"` denotes mandatory.

if can you point to it that could be useful although likely unneeded

Reply:
> You could just copy unknown tag attrs to the HTML tag. There is some Jelly/Stapler trick to do that.

Second this as it is important for my use case.

Reply:
https://github.com/jenkinsci/jenkins/blob/7c1885554e394176dacd6296236403a3181ef351/core/src/main/java/hudson/util/jelly/MorphTagLibrary.java#L24-L32 https://github.com/jenkinsci/jenkins/blob/7c1885554e394176dacd6296236403a3181ef351/core/src/main/resources/lib/form/textbox.jelly#L107-L113

Reply:
Thanks, @strangelookingnerd what is missing that you need?

Reply:
See https://github.com/jenkinsci/custom-folder-icon-plugin/blob/master/src/main/resources/jenkins/plugins/foldericon/CustomFolderIcon/config.jelly#L34

I'm adapting the size (for visual reasons) and add an ID to add a change event in the following `<script>` section 
https://github.com/jenkinsci/custom-folder-icon-plugin/blob/5f60c7af85f7dfd56caca7e6ce392e70442ab616/src/main/resources/jenkins/plugins/foldericon/CustomFolderIcon/config.jelly#L61-L69

Maybe this is the wrong approach but without the ID my use case breaks.

Reply:
Styling should be done via classes and not inline css, I added support for an ID because of your plugin but you could also do that via a CSS class anyway